docs/pvh.md

```diff
@@ -0,0 +1,15 @@
+# PVH boot mode
+
+Firecracker supports booting x86 kernels in "PVH direct boot" mode
+[as specified by the Xen project](https://github.com/xen-project/xen/blob/master/docs/misc/pvh.pandoc).
+If a kernel is provided which contains the XEN_ELFNOTE_PHYS32_ENTRY ELF Note
+then this boot mode will be used.  This boot mode was designed for virtualized
+environments which load the kernel directly, and is simpler than the "Linux
+boot" mode which is designed to be launched from a legacy boot loader.
+
+PVH boot mode can be enabled for Linux by setting CONFIG_XEN_PVH=y in the
+kernel configuration.  (This is not the default setting.)
+
+PVH boot mode is enabled by default in FreeBSD, which has support for
+Firecracker starting with FreeBSD 14.0.  Instructions on building a FreeBSD
```

Comment:
Is FreeBSD support for Firecracker correlated to the Firecracker support of booting FreeBSD?

Reply:
Yes.  FreeBSD cannot boot on non-PVH Firecracker, but with these changes FreeBSD is able to boot on Firecracker.

Reply:
yes, with this patch FC can boot FreeBSD. It is just the part `which(FreeBSD I assume) has support for Firecracker starting with FreeBSD 14.0` is confusing. Is PVH enabled by default in FreeBSD only from 14.0 version?

Reply:
Oh, FreeBSD has had PVH support for longer than that, but wouldn't boot in Firecracker/PVH due to other issues (assuming that PVH booting = support for Xen hypercalls, lacking virtio mmio cmdline support, etc).  So it really is specifically Firecracker support which landed in FreeBSD 14.0.

Reply:
Ok now it makes sense, thx for the explanation.